cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -169,6 +183,145 @@ protected void setMetricsEnabled(boolean isMetricsEnabled) {
         this.isMetricsEnabled = isMetricsEnabled;
     }
 
+    /**
+     * Returns logging configuration. In case of no changes default logging configuration is returned.
+     * @return
+     */
+    protected String getLogConfig() {
+        return logConfig;
+    }
+
+    public void setLog(String logCfg) {
+        this.logConfig = logCfg;
+    }
+
+    /**
+     * Returns map with all available loggers for current pod and default values.
+     * @return
+     */
+    abstract protected Properties getDefaultLogConfig();
+
+    protected Properties abstractGetDefaultLoggingConfig(String defaultConfigResourceFileName) {
+        Properties defaultSettings = new Properties();
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        InputStream is = classloader.getResourceAsStream(defaultConfigResourceFileName);
+        try (Scanner scanner = new Scanner(is, "UTF-8")) {
+
+            while (scanner.hasNextLine()) {
+                String line = scanner.nextLine();
+                if (!line.contains("=")) {
+                    log.warn("Assignment missing");
+                    break;
+                }
+                defaultSettings.put(line.split("=")[0], line.split("=")[1]);
+            }
+            scanner.close();
+        }
+        return defaultSettings;
+    }
+
+    /**
+     * Transforms map to log4j properties file format
+     * @param newSettings map with properties
+     * @return
+     */
+    protected static String createPropertiesString(Properties newSettings) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("# Do not change this generated file. To change logging settings please use cluster config map.\n");
+        newSettings.forEach((key, value) -> {
+            sb.append(key);
+            sb.append("=");
+            sb.append(value);
+            sb.append("\n");
+        });
+        return sb.toString();
+    }
+
+    /**
+     * Receives Json object of custom settings. If Json contains "fromConfigMap" custom config map is used.
+     * Otherwise single fields are parsed and validated. From these fields is created log4j-like string.
+     * @param logConfig
+     */
+    protected void parseAndSetLogging(JsonObject logConfig) {
+        Map<String, Object> origin;
+        Properties newSettings;
+
+        if (logConfig == null) {
+            // field is not in the cluster CM
+            newSettings = getDefaultLogConfig();
+            this.logConfig = createPropertiesString(newSettings);
+        } else {
+            origin = logConfig.getMap();
+            if (origin.get("fromConfigMap") != null) {
```

Comment:
To me `fromConfigMap` is acting as a discriminator to distinguish two subclasses: `ExternalConfigMapLogging` and `InlineLogging`, both extending a common `abstract Logging` class. The CRDs use Jackson to marshal YAML/JSON to Java objects. So in this PR you should write those classes. This class would then have a `Logging`-typed field instead of the `logConfig` and `customerCMName` fields, and this method could be replace by a plain `void setLogging(Logging)` (the conversion to a properties String could be deferred to `generateMetricsAndLogConfigMap()`